### PR TITLE
Fix Accepting Submissions Bug

### PR DESCRIPTION
**Pull request by aguywithanidea**

### CODE REVIEW

File: include/classes/HeyPublisher/Page/Options.class.php [hunk 1]
```diff
@@ -459,7 +459,8 @@ private function merged_genre_map($my_genres) {
       $accumulator[$id] = ($item['wp_id']) ? $item['wp_id'] : $saved_genres["$id"];
       return $accumulator;
     });
-    $test = array_filter($has);
+    $test = array();
```
Comment (aguywithanidea) [inline include/classes/HeyPublisher/Page/Options.class.php:462]: accounting for edge-case where this var is null

File: include/classes/HeyPublisher/Page/Options.class.php [hunk 2]
```diff
@@ -735,54 +737,56 @@ private function validate_user($post) {
       $pub = $this->pubapi->get_publisher_info();
       $message = 'Account validation succeeded! You can now configure your account.';
       if ($pub) {
+        // TODO: Change this to pull from JSON endpoint
```
Comment (aguywithanidea) [inline include/classes/HeyPublisher/Page/Options.class.php:740]: Notes for next time

File: include/classes/HeyPublisher/Page/Options.class.php [hunk 3]
```diff
@@ -735,54 +737,56 @@ private function validate_user($post) {
...
         $cats = $this->xml->get_my_categories_as_hash();
-        $has_genres = '0';
+        $has_genres = false;
```
Comment (aguywithanidea) [inline include/classes/HeyPublisher/Page/Options.class.php:742]: the var being set is a boolean, so we should be consistent here

File: include/classes/HeyPublisher/Page/Options.class.php [hunk 4]
```diff
@@ -735,54 +737,56 @@ private function validate_user($post) {
...
     // Get options from the post
     $opts = $post['heypub_opt'];
     $this->logger->debug(sprintf("\t\$opts = %s",print_r($opts,1)));
-    // Need to delect the values in category_map where:
+    // Need to detect the values in category_map where:
     //  a) value is < 0
     //  b) key is not present in genres array
     $this->clean_genres_category_map($opts);
+    // $this->logger->debug(sprintf("\tabout to test accepting subs = %s\n\tactive = ",print_r($opts['category_map'],1),$opts['active']));
+    $opts['accepting_subs'] = false;
```
Comment (aguywithanidea) [inline include/classes/HeyPublisher/Page/Options.class.php:779]: This fixes the bug

**Review by aguywithanidea — COMMENTED** [review]
fixing bug where it's impossible to unset the "accepting submissions" boolean once set.

**Comment by aguywithanidea** [comment]
The PR fixes bug #110
